EngineLargeStatePerformanceTest.shouldProcessWithinExpectedDeviation is unstable

**Summary**

- How often does the test fail? 1
- Does it block your work? 
- Do we suspect that it is a real failure? 

**Failures**

`EngineLargeStatePerformanceTest.shouldProcessWithinExpectedDeviation:144 Expected reference score is "450.000" got "334.123", deviation "25.751%" exceeds maximum allowed deviation "15.000%"`

**Logs**

Downloads complete logs from [here](https://drive.google.com/drive/folders/1R8X_FNT6siSQtIydy9voIv1XxJp_tVaZ)

Failed again with: https://github.com/camunda/zeebe/actions/runs/6971982216/job/18973214307

```
[INFO] Running io.camunda.zeebe.engine.perf.EngineLargeStatePerformanceTest
Error:  Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 332.7 s <<< FAILURE! -- in io.camunda.zeebe.engine.perf.EngineLargeStatePerformanceTest
Error:  io.camunda.zeebe.engine.perf.EngineLargeStatePerformanceTest.shouldProcessWithinExpectedDeviation(JMHTestCase) -- Time elapsed: 332.7 s <<< FAILURE!
java.lang.AssertionError: Expected reference score is "450.000" got "304.843", deviation "32.257%" exceeds maximum allowed deviation "15.000%"
	at io.camunda.zeebe.engine.perf.EngineLargeStatePerformanceTest.shouldProcessWithinExpectedDeviation(EngineLargeStatePerformanceTest.java:144)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```


On main when it succeeds https://github.com/camunda/zeebe/actions/runs/6979181397/job/18991962899
```
Benchmark                                                  Mode  Cnt   Score   Error  Units
LargeStateControllerPerformanceTest.measureStateRecovery  thrpt   25  13.583 ± 0.102  ops/s

Benchmark                                                     Mode  Cnt    Score   Error  Units
EngineLargeStatePerformanceTest.measureProcessExecutionTime  thrpt   50  509.332 ± 5.992  ops/s
```

or https://github.com/camunda/zeebe/actions/runs/6946211264
```
Benchmark                                                  Mode  Cnt   Score   Error  Units
LargeStateControllerPerformanceTest.measureStateRecovery  thrpt   25  12.388 ± 0.181  ops/s

Benchmark                                                     Mode  Cnt    Score   Error  Units
EngineLargeStatePerformanceTest.measureProcessExecutionTime  thrpt   50  431.323 ± 4.483  ops/s
```

How?!

Failed again on [https://github.com/camunda/zeebe/actions/runs/7057426514/job/19211087404?pr=15443](https://github.com/camunda/zeebe/actions/runs/7057426514/job/19211087404?pr=15443)

```
java.lang.AssertionError: Expected reference score is "450.000" got "522.296", deviation "16.066%" exceeds maximum allowed deviation "15.000%"
```

Another time in [25731730569](https://github.com/camunda/camunda/actions/runs/9349739713/job/25731730569) for attempting to merge https://github.com/camunda/camunda/pull/18894#event-13020095018

>java.lang.AssertionError: Expected reference score is "450.000" got "528.414", deviation >"17.425%" exceeds maximum allowed deviation "15.000%"

--- a/engine/src/test/java/io/camunda/zeebe/engine/perf/EngineLargeStatePerformanceTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/perf/EngineLargeStatePerformanceTest.java
@@ -141,6 +141,6 @@ public class EngineLargeStatePerformanceTest {
     final var assertResult = testCase.run();
 
     // then
-    assertResult.isWithinDeviation(referenceScore, 0.15);
+    assertResult.isAtLeast(referenceScore, 0.15);
   }
 }

--- a/engine/src/test/java/io/camunda/zeebe/engine/perf/EngineLargeStatePerformanceTest.java
+++ b/engine/src/test/java/io/camunda/zeebe/engine/perf/EngineLargeStatePerformanceTest.java
@@ -141,6 +141,6 @@ public class EngineLargeStatePerformanceTest {
     final var assertResult = testCase.run();
 
     // then
-    assertResult.isAtLeast(referenceScore, 0.15);
+    assertResult.isAtLeast(referenceScore, 0.2);
   }
 }

